src/query/sql/src/executor/physical_plans/physical_merge_into_add_row_number.rs

```diff
@@ -1,80 +0,0 @@
-// Copyright 2021 Datafuse Labs
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-use std::collections::BTreeMap;
-use std::sync::Arc;
-
-use databend_common_exception::ErrorCode;
-use databend_common_exception::Result;
-use databend_common_expression::types::DataType;
-use databend_common_expression::types::NumberDataType;
-use databend_common_expression::DataField;
-use databend_common_expression::DataSchema;
-use databend_common_expression::DataSchemaRef;
-use databend_common_expression::ROW_NUMBER_COL_NAME;
-
-use crate::executor::PhysicalPlan;
-use crate::executor::PhysicalPlanBuilder;
-use crate::optimizer::ColumnSet;
-use crate::optimizer::SExpr;
-
-// add row_number for distributed merge into
-#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct MergeIntoAddRowNumber {
```

Comment:
If there is a broadcast join, we cannot keep the insert data in the local node, we need to shuffle the source table after the join, which requires the `row_number` and some deduplication operations.